core/src/main/java/com/graphhopper/reader/osm/OSMNodeData.java

```diff
@@ -258,9 +253,9 @@ public Map<String, Object> getTags(long osmNodeId) {
         return nodeTags.get(tagIndex);
     }
 
-    public void removeTags(long osmNodeId) {
+    public void removeTag(long osmNodeId, String key) {
         int prev = nodeTagIndicesByOsmNodeIds.put(osmNodeId, -2);
```

Comment:
Why do you think it is no longer intended? Removing the tag is necessary to split a node multiple times. But maybe this case never occurs, because we do not allow splitting junction nodes at all: https://github.com/graphhopper/graphhopper/blob/70ea1ca9c05f2b7a0f320809874188c66c9025fc/core/src/main/java/com/graphhopper/reader/osm/WaySegmentParser.java#L228-L235

Reply:
Ah, I see. Yes this makes no sense. We need something like this (?)

```java
        int tagIndex = nodeTagIndicesByOsmNodeIds.get(osmNodeId);
        if (tagIndex >= 0)
            nodeTags.get(tagIndex).remove(key);
        if (nodeTags.isEmpty())
            nodeTagIndicesByOsmNodeIds.put(osmNodeId, -2);
 ```

Reply:
... but if nodes tagged as `gh:split_node` can never occur at junctions, maybe we do not need this at all.

Reply:
... no, we need a way to recognize which nodes were already split, because the barrier nodes can also occur at the ends of OSM ways, connecting two ways.